backend/de.metas.payment.esr/src/main/java/de/metas/payment/esr/api/impl/ESRBL.java

```diff
@@ -61,37 +44,55 @@
 import de.metas.util.Services;
 import de.metas.util.StringUtils;
 import lombok.NonNull;
+import org.adempiere.model.InterfaceWrapperHelper;
+import org.adempiere.util.lang.IContextAware;
+import org.adempiere.util.lang.impl.TableRecordReference;
+import org.compiere.model.I_AD_Org;
+import org.compiere.model.I_C_Invoice;
+import org.compiere.model.MOrg;
+import org.compiere.util.Env;
+import org.slf4j.Logger;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.List;
+import java.util.Properties;
+
+import static org.adempiere.model.InterfaceWrapperHelper.newInstance;
+import static org.adempiere.model.InterfaceWrapperHelper.saveRecord;
 
 public class ESRBL implements IESRBL
 {
 	private final transient Logger logger = LogManager.getLogger(getClass());
+	private final IInvoiceBL invoiceBL = Services.get(IInvoiceBL.class);
 
 	@Override
 	public boolean appliesForESRDocumentRefId(final Object sourceModel)
 	{
-		final String sourceTableName = InterfaceWrapperHelper.getModelTableName(sourceModel);
+		String sourceTableName = InterfaceWrapperHelper.getModelTableName(sourceModel);
 		Check.assume(I_C_Invoice.Table_Name.equals(sourceTableName), "Document " + sourceModel + " not supported");
 
 		final I_C_Invoice invoice = InterfaceWrapperHelper.create(sourceModel, I_C_Invoice.class);
 
-		if (!invoice.isSOTrx())
-		{
-			logger.debug("Skip generating because invoice is purchase invoice: {}", invoice);
-			return false;
-		}
-		if (invoice.getReversal_ID() > 0)
-		{
-			logger.debug("Skip generating because invoice is a reversal: {}", invoice);
-			return false;
-		}
-		if (Services.get(IInvoiceBL.class).isCreditMemo(invoice))
-		{
-			logger.debug("Skip generating because invoice is a credit memo: {}", invoice);
-			return false;
-		}
-		return true;
+		return isInvoiceOfInterest(invoice);
+	}
+
+	private boolean isInvoiceOfInterest(I_C_Invoice invoice)
```

Comment:
pls inline isInvoiceOfInterest method. sounds very weird :)

Reply:
not to mention that the "appliesForESRDocumentRefId" method also suffers from the "ombelico del mondo" syndrome.